core/arch/arm/kernel/secure_partition.c

```diff
@@ -1418,111 +1418,76 @@ static const struct ts_ops sp_ops = {
 
 static TEE_Result process_sp_pkg(uint64_t sp_pkg_pa, TEE_UUID *sp_uuid)
 {
-	enum teecore_memtypes mtype = MEM_AREA_RAM_SEC;
+	enum teecore_memtypes mtype = MEM_AREA_TA_RAM;
 	struct sp_pkg_header *sp_pkg_hdr = NULL;
-	TEE_Result res = TEE_SUCCESS;
-	tee_mm_entry_t *mm = NULL;
 	struct fip_sp *sp = NULL;
 	uint64_t sp_fdt_end = 0;
 	size_t sp_pkg_size = 0;
 	vaddr_t sp_pkg_va = 0;
-	size_t num_pages = 0;
 
-	/* Map only the first page of the SP package to parse the header */
+	/* Check the first page which contains the SP package header */
 	if (!tee_pbuf_is_sec(sp_pkg_pa, SMALL_PAGE_SIZE))
```

Comment:
This isn't needed, `phys_to_virt()` below will fail if the PA isn't in the expected range.

Reply:
Thanks, fixed.